azure.pkr.hcl

```diff
@@ -87,13 +87,21 @@ variable "packer_work_group" {
   description = "The resource group for Packer to use while building the VM"
 }
 
+locals {
+  # Input is either:
+  # - 20_04-daily-lts-gen2
+  # - 20_04-daily-lts-arm64
+  # The output should be either x64 or arm64
+  image_architecture = replace(replace(var.source_image_sku, "20_04-daily-lts-", ""), "gen2", "x64")
```

Comment:
Maybe we should flip this, and allow an architecture to be specified that we then use to choose the `source_image_sku`? That way we know we're not going to generate a garbage architecture if an image is passed with the wrong format.

If so the same comment applies to GCP.